config/locales/en/devise.yml

```diff
@@ -15,6 +15,8 @@ en:
     omniauth_callbacks:
       failure: "Could not authenticate you from %{kind} because \"%{reason}\"."
       success: "Successfully authenticated from %{kind} account."
+      provider_gatekeeper:
+        no_office_codes: "Your account cannot use this service as it does not have any office codes."
```

Comment:
Hehe, yep it may be a bit obscure, but this error is only for when the user doesn't have any office accounts at all. This, to the best of my understanding, should not happen in production, but it is happening with some accounts (in SAMLmock for sure, maybe dev/staging envs too), rendering our service useless as applications need to always be linked to a provider's office account code.

The enrolment/authorisation I believe you are thinking about (the one based on office codes, firms (if we have all codes for a given firm), or even down to individual email addresses) will come next, it is not done in this PR, and will have different error message(s) 😄

Reply:
Right, yep, makes sense, thanks for explaining!!